### PR TITLE
Update gsclient-go v1.0 compatibility & reformat server code

### PR DESCRIPTION
**Pull request by nvthongswansea**

### CODE REVIEW

File: gridscale/service-query/blocker.go [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+	IPService           gsService = "IP"
+	NetworkService      gsService = "network"
+	ServerService       gsService = "server"
+	SSHKeyService       gsService = "sshkey"
+	StorageService      gsService = "storage"
+	ISOImageService     gsService = "isoimage"
+)
+
+//BlockProvisoning blocks until the object's state is not in provisioning anymore
+func BlockProvisoning(client *gsclient.Client, service gsService, id string, timeout time.Duration) error {
```
Comment (itakouna) [inline gridscale/service-query/blocker.go:33]: I would prefer `RetryUntilResourceStatusIsActive` to reflect what the function does.

File: gridscale/service-query/blocker.go [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+	return resource.Retry(timeout, func() *resource.RetryError {
+		time.Sleep(delayFetchingStatus)
+		switch service {
+		case LoadbalancerService:
+			lb, err := client.GetLoadBalancer(id)
+			if err != nil {
+				return resource.NonRetryableError(fmt.Errorf(
+					"Error waiting for loadbalancer (%s) to be fetched: %s", id, err))
```
Comment (itakouna) [inline gridscale/service-query/blocker.go:41]: errors with status error >= 500 should be `RetryableError`

File: gridscale/service-query/blocker.go [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+	ServerService       gsService = "server"
+	SSHKeyService       gsService = "sshkey"
+	StorageService      gsService = "storage"
+	ISOImageService     gsService = "isoimage"
+)
+
+//BlockProvisoning blocks until the object's state is not in provisioning anymore
+func BlockProvisoning(client *gsclient.Client, service gsService, id string, timeout time.Duration) error {
+	return resource.Retry(timeout, func() *resource.RetryError {
+		time.Sleep(delayFetchingStatus)
```
Comment (itakouna) [inline gridscale/service-query/blocker.go:35]: I do not think `time.Sleep(delayFetchingStatus)`  is needed. the retry will be triggered after this time `timeout` and forever. I think we need a way to stop the retry after awhile e.g, 10 minutes.

File: gridscale/resource_gridscale_loadbalancer.go [hunk 4]
```diff
@@ -189,30 +189,37 @@ func resourceGridscaleLoadBalancerUpdate(d *schema.ResourceData, meta interface{
...
+	err = service_query.BlockProvisoning(client, service_query.LoadbalancerService, d.Id(), d.Timeout(schema.TimeoutUpdate))
+	if err != nil {
+		return err
+	}
 	return resourceGridscaleLoadBalancerRead(d, meta)
 }
 
 func resourceGridscaleLoadBalancerDelete(d *schema.ResourceData, meta interface{}) error {
 	client := meta.(*gsclient.Client)
-	id := d.Id()
```
Comment (itakouna) [inline gridscale/resource_gridscale_loadbalancer.go:214]: what do you think about this code:

```
return resource.Retry(d.Timeout(schema.TimeoutDelete), func() *resource.RetryError {
		err := client.DeleteLoadBalancer(id)

		if err != nil {
			if requestError, ok := err.(gsclient.RequestError); ok {
				if requestError.StatusCode == 424 || requestError.StatusCode >= 500 {
					return resource.RetryableError(fmt.Errorf("Error getting loadbalancer: %s", err))
				} else{
					return resource.NonRetryableError(fmt.Errorf("Error getting loadbalancer: %s", err))
				}
			}
		}

		return resource.NonRetryableError(err)
	})

File: gridscale/resource_gridscale_loadbalancer.go [hunk 5]
```diff
@@ -189,30 +189,37 @@ func resourceGridscaleLoadBalancerUpdate(d *schema.ResourceData, meta interface{
...
-	return client.DeleteLoadBalancer(id)
+	err := resource.Retry(d.Timeout(schema.TimeoutDelete), func() *resource.RetryError {
+		return resource.RetryableError(client.DeleteLoadBalancer(d.Id()))
+	})
+	if err != nil {
+		return err
+	}
+	return service_query.BlockDeletion(client, service_query.LoadbalancerService, d.Id())
```
Comment (itakouna) [inline gridscale/resource_gridscale_loadbalancer.go:222]: I do not prefer using `resource.Retry` and `BlockDeletion` in this way.

File: gridscale/resource_gridscale_loadbalancer.go (hunk 4 above)
Comment (nvthongswansea) [reply gridscale/resource_gridscale_loadbalancer.go:214]: Could you explain please?

File: gridscale/service-query/blocker.go (hunk 3 above)
Comment (nvthongswansea) [reply gridscale/service-query/blocker.go:35]: I tried the code you gave me and it stops after the timeout.
`
return resource.Retry(10*time.Second, func() *resource.RetryError {
		time.Sleep(1 * time.Second)
		err := client.DeleteLoadBalancer(d.Id())
		return resource.RetryableError(fmt.Errorf("Error getting loadbalancer: %s", err))
	})
`

Comment (nvthongswansea) [reply gridscale/service-query/blocker.go:35]: I still keep `time.Sleep` as a rate limiter like `WaitForRequestCompletion` and `WaitForServerPowerStatus` in `gsclient-go`.

**Comment by nvthongswansea** [comment]
I close this PR because I want to separate it into smaller PRs.

**Pull request closed without merging**